spike_main/spike.cc

```diff
@@ -123,31 +123,48 @@ bool sort_mem_region(const mem_cfg_t &a, const mem_cfg_t &b)
     return (a.base < b.base);
 }
 
-void merge_overlapping_memory_regions(std::vector<mem_cfg_t> &mems)
+static bool check_mem_overlap(const mem_cfg_t& L, const mem_cfg_t& R)
```

Comment:
Maybe make this a member function on `mem_cfg_t`, so that below you can write something like `assert(!L.overlaps(R))`

Reply:
Leaving it as-is is OK with me.